src/main/java/homework_2/random_chars_table/RandomCharsTable.java

```diff
@@ -0,0 +1,104 @@
+package homework_2.random_chars_table;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class RandomCharsTable {
+
+    public static final String ERROR = "Passed parameters should match the format [positive integer] [positive integer] [even|odd]";
+
+    private int length;
+    private int width;
+    private String strategy;
+
+    private String strategyEven = "Even letters -";
+    private String strategyOdd = "Odd letters -";
+
+    public void run() {
+
+        System.out.println("[Input example: 2 3 odd]");
+        System.out.println("Please, input 2 positive numbers and a strategy (even/odd)");
+
+        String inputFull = bufferedReaderReadConsole();
+        try {
+            parseInput(inputFull);
+            validateInputData(length, width, strategy);
+        } catch (IllegalArgumentException e) {
+            System.out.println(ERROR);
+            return;
+        }
+
+        char[][] table = new char[length][width];
+
+        for (int i = 0; i < length; i++) {
+            System.out.print("|");
+
+            for (int j = 0; j < width; j++) {
+                int x = (int) (Math.random() * 26 + 65);
+                table[i][j] = (char) x;
+                if (x % 2 == 0) {
+                    strategyEven = strategyEven + " " + table[i][j] + ",";
+                } else {
+                    strategyOdd = strategyOdd + " " + table[i][j] + ",";
+                }
+                System.out.print(" " + table[i][j] + " |");
+            }
+
+            System.out.println();
+        }
+
+        strategyOdd = deleteLastCommaInStrategy(strategyOdd);
+        strategyEven = deleteLastCommaInStrategy(strategyEven);
+
+        if (strategy.equals("even")) {
+            System.out.print(strategyEven);
+        }
+
+        if (strategy.equals("odd")) {
+            System.out.print(strategyOdd);
+        }
+    }
+
+    private String bufferedReaderReadConsole() {
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(System.in))) {
+            String input = reader.readLine();
+            return input;
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        // Корректно ли здесь возвращать null или нет?
+        return null;
```

Comment:
In this case upon returning null and assigning it to inputFull in line 23 you risk getting NullPointerExc in line 76 since you haven't checked s for null.